### PR TITLE
Rstgen/xml tree fixes

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

**Comment by stisa** [comment]
Note that removing `lang` prefix will break `run` on forum due to https://github.com/nim-lang/nimforum/commit/a44e17d03a0c2de022113a24a92fa39e40b5c56b

**Comment by dom96** [comment]
I know :)

File: lib/pure/xmltree.nim [hunk 1]
```diff
@@ -316,7 +319,8 @@ proc xmlConstructor(a: NimNode): NimNode {.compileTime.} =
     var elements = newNimNode(nnkBracket, a)
     for i in 1..a.len-1:
       if a[i].kind == nnkExprEqExpr:
-        attrs.add(toStrLit(a[i][0]))
+        let attrName = toStrLit(a[i][0]).strVal.replace(" ", "")
```
Comment (Araq) [inline lib/pure/xmltree.nim]: Why is this required? Add a comment.
